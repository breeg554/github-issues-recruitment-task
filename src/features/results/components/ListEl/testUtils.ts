export const fakeRepoLanguage = {
  name: "test",
  color: "#fff",
  __typename: "language",
};

export const fakeUserData = {
  avatarUrl:
    "https://avatars.githubusercontent.com/u/499550?u=de41ec9325e8a92e281b96a1514a0fd1cd81ad4a&v=4",
  bio: "",
  email: "",
  id: "MDQ6VXNlcjQ5OTU1MA==",
  location: "Singapore",
  login: "yyx990803",
  name: "Evan You",
  __typename: "User",
};
export const fakeRepoData = {
  description: "Monkey testing library for web apps and Node.js",
  id: "MDEwOlJlcG9zaXRvcnkxMzczNzE0OQ==",
  issues: { totalCount: 6, __typename: "IssueConnection" },
  languages: {
    nodes: [
      {
        color: "#f1e05a",
        name: "JavaScript",
        __typename: "Language",
      },
    ],
    __typename: "LanguageConnection",
  },
  licenseInfo: {
    name: "MIT License",
    __typename: "License",
  },
  name: "gremlins.js",
  stargazerCount: 8798,
  updatedAt: "2022-01-01T04:37:32Z",
  url: "https://github.com/marmelab/gremlins.js",
  __typename: "Repository",
};
