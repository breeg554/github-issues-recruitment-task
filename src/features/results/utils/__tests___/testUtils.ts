export const fakeResponseData = {
  users: {
    dataCount: 10,
    edges: [
      {
        node: {},
      },
      {
        node: {},
      },
    ],
    pageInfo: {
      endCursor: null,
      hasNextPage: false,
      hasPreviousPage: false,
      startCursor: null,
    },
  },
  repositories: {
    dataCount: 0,
    edges: [
      {
        node: {},
      },
      {
        node: {},
      },
    ],
    pageInfo: {
      endCursor: null,
      hasNextPage: false,
      hasPreviousPage: false,
      startCursor: null,
    },
  },
};
