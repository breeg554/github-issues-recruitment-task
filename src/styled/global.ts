import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html,button,input,textarea{
    font-family: 'Roboto', sans-serif;
    font-size: 10px;   //1rem = 10px
  }
  body{
    background-color: #E5E5E5;
  }
`;
