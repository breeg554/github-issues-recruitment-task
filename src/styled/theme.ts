export type ThemeType = typeof theme; // This is the type definition for my theme object.

export const theme = {
  colors: {
    core: {
      primary: "#24292F",
      secondary: "#166CD7",
      tertiary: "#6F7781",
    },
    hover: {
      primary: "#353a3F",
      secondary: "#288ef9",
      tertiary: "#9F9992",
    },
  },
  mediaQ: {
    sm: "@media(min-width: 576px)",
    md: "@media(min-width: 768px)",
    lg: "@media(min-width: 992px)",
    xl: "@media(min-width: 1200px)",
  },
};
