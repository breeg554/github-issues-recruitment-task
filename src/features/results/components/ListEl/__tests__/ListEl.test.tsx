import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { ListEl } from "..";
import { theme } from "../../../../../styled/theme";
import { fakeRepoData, fakeUserData } from "../testUtils";
describe("List element component", () => {
  it("renders out user data and match the snapshot", () => {
    const { container } = render(
      <Router>
        <ThemeProvider theme={theme}>
          <ListEl data={fakeUserData} />
        </ThemeProvider>
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
  it("renders out repository data and match the snapshot", () => {
    const { container } = render(
      <Router>
        <ThemeProvider theme={theme}>
          <ListEl data={fakeRepoData} />
        </ThemeProvider>
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
