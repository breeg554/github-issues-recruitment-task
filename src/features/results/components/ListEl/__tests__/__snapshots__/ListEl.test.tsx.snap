// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List element component renders out repository data and match the snapshot 1`] = `
<div>
  <li
    class="sc-bdvvtL eSrCGW"
  >
    <article>
      <span
        class="repo-icon"
      >
        <svg
          aria-hidden="true"
          class="octicon octicon-repo"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div
        class="site-wrapper"
      >
        <header
          class="sc-gsDKAQ ktjAct"
        >
          <a
            href="/"
          >
            <h2>
              gremlins.js
            </h2>
          </a>
          <p
            class="description"
          >
            Monkey testing library for web apps and Node.js
          </p>
        </header>
        <footer
          class="sc-dkPtRN gcNZJC"
        >
          <span
            class="sc-furwcr gXfInY"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-star"
              fill="currentColor"
              height="14"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 16 16"
              width="14"
            >
              <path
                d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                fill-rule="evenodd"
              />
            </svg>
            8798
          </span>
          <span
            class="sc-hKwDye gMbWyd"
          >
            <span />
            JavaScript
          </span>
          <span>
            MIT License
          </span>
          <span>
            a day ago
          </span>
          <span>
            6
             
            issues
             needs help
          </span>
        </footer>
      </div>
    </article>
  </li>
</div>
`;

exports[`List element component renders out user data and match the snapshot 1`] = `
<div>
  <li
    class="sc-bdvvtL eSrCGW"
  >
    <article>
      <a
        href="/yyx990803"
      >
        <span
          class="sc-pVTFL iDLXbb"
          width="20px"
        >
          <img
            alt="yyx990803"
            src="https://avatars.githubusercontent.com/u/499550?u=de41ec9325e8a92e281b96a1514a0fd1cd81ad4a&v=4"
            style="opacity: 0;"
          />
          <span
            class="icon"
            style="visibility: visible;"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-mark-github"
              fill="currentColor"
              height="16"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
      </a>
      <div
        class="site-wrapper"
      >
        <header
          class="sc-gsDKAQ ktjAct"
        >
          <a
            href="/yyx990803"
          >
            <h2>
              Evan You
            </h2>
          </a>
          <p
            class="description"
          >
            yyx990803
          </p>
        </header>
        <footer
          class="sc-dkPtRN gcNZJC"
        >
          <span>
            Singapore
          </span>
        </footer>
      </div>
    </article>
  </li>
</div>
`;
